RUFAS/report_generator.py

```diff
@@ -0,0 +1,292 @@
+    return sum(data) / len(data) if data else 0
+
+
+def division_aggregator(data: List[float]) -> float:
+    """
+    Divides the first number in the list by each of the subsequent numbers.
+
+    Parameters
+    ----------
+    data : List[float]
+        A list of numbers for the division operation.
+
+    Returns
+    -------
+    float
+        The result of dividing the first number by each subsequent number.
+        Returns None if the list is empty or has only one element.
+    """
+    if len(data) < 2:
+        return None
+    result = data[0]
+    for num in data[1:]:
+        if num == 0:  # Avoid division by zero
+            return None
+        result /= num
+    return result
+
+
+def product_aggregator(data: List[float]) -> float:
+    """
+    Returns the product of a list of numbers.
+
+    Parameters
+    ----------
+    data : List[float]
+        A list of numbers whose product is to be calculated.
+
+    Returns
+    -------
+    float
+        The product of the input numbers. Returns 1 for an empty list.
+    """
+    product = 1
+    for num in data:
+        product *= num
+    return product
+
+
+def sd_aggregator(data: List[float]) -> float:
+    """
+    Calculates the standard deviation of a list of numbers.
+
+    Parameters
+    ----------
+    data : List[float]
+        A list of numbers whose standard deviation is to be calculated.
+
+    Returns
+    -------
+    float
+        The standard deviation of the input numbers.
+    """
+    mean = average_aggregator(data)
+    return (sum((x - mean) ** 2 for x in data) / len(data)) ** 0.5 if data else 0
+
+
+def sum_aggregator(data: List[float]) -> float:
+    """
+    Returns the sum of a list of numbers.
+
+    Parameters
+    ----------
+    data : List[float]
+        A list of numbers whose sum is to be calculated.
+
+    Returns
+    -------
+    float
+        The sum of the input numbers.
+    """
+    return sum(data)
+
+
+def subtraction_aggregator(data: List[float]) -> float:
+    """
+    Subtracts each subsequent number in the list from the first number.
+
+    Parameters
+    ----------
+    data : List[float]
+        A list of numbers for the subtraction operation.
+
+    Returns
+    -------
+    float
+        The result of subtracting each subsequent number from the first number.
+        Returns None if the list is empty or has only one element.
+    """
+    if len(data) < 2:
+        return None
+    result = data[0]
+    for num in data[1:]:
+        result -= num
+    return result
+
+
+AGGREGATION_FUNCTIONS: Dict[str, Callable[[List[float]], float]] = {
+    "average": average_aggregator,
+    "division": division_aggregator,
+    "product": product_aggregator,
+    "SD": sd_aggregator,
+    "sum": sum_aggregator,
+    "subtraction": subtraction_aggregator
+}
+
+
+class ReportGenerator:
+    def generate_report(
+        self,
+        filtered_pool: Dict[str, Dict[str, List[Any]]],
+        filter_content: Dict[str, str | int],
```

Comment:
The key `"variables"` in this argument is supposed to point to a list of strings, if I understand the code correctly.
```suggestion
        filter_content: Dict[str, str | int | List[str]],
```